collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceGoogleMapActivity.java

```diff
@@ -505,7 +505,7 @@ private void setupAutomaticMode() {
         String units = timeUnits.getSelectedItem().toString();
         Long timeDelay;
         TimeUnit timeUnitsValue;
-        if (units == getString(R.string.minutes)) {
+        if (units.equals(getString(R.string.minutes))) {
```

Comment:
Since this could never have been true before, and now can be, have you exercised the true path?

Reply:
== will still return true when Java pools Strings which seem like was the case in this scenario but String pooling is not a definitive behaviour and can not be guaranteed. I did exercise the true path and it worked as expected

Reply:
Ah, thanks for the reminder. I had forgotten that.